From ab756b13392448201c106b7629c55f68af4d8ec5 Mon Sep 17 00:00:00 2001
From: Adam Jackson <ajax@redhat.com>
Date: Fri, 1 Oct 2021 11:47:21 -0400
Subject: [PATCH xserver] mustard: xfree86: Only call the driver's
 platformProbe

The code to detangle what's a PCI device and what's platform is... bad.
The PCI-specific initialization path should never be necessary these
days, and the ancient "just probe something" path should especially
never be necessary, so we can avoid a bunch of fragile code by simply
never calling into the non-platform probe paths and letting platform
handle everything.

Among other things this fixes hypervdrm on v1 devices, which have both a
PCI and a platform presentation, in a way that would cause X to fail to
start.
---
 hw/xfree86/common/xf86Bus.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/hw/xfree86/common/xf86Bus.c b/hw/xfree86/common/xf86Bus.c
index fd144dbe7a..88e07dec40 100644
--- a/hw/xfree86/common/xf86Bus.c
+++ b/hw/xfree86/common/xf86Bus.c
@@ -84,6 +84,7 @@ xf86CallDriverProbe(DriverPtr drv, Bool detect_only)
     }
 #endif
 
+#if 0
 #ifdef XSERVER_LIBPCIACCESS
     if (!foundScreen && (drv->PciProbe != NULL)) {
         if (xf86DoConfigure && xf86DoConfigurePass1) {
@@ -102,6 +103,7 @@ xf86CallDriverProbe(DriverPtr drv, Bool detect_only)
         foundScreen = (*drv->Probe) (drv, (detect_only) ? PROBE_DETECT
                                      : PROBE_DEFAULT);
     }
+#endif
 
     return foundScreen;
 }
-- 
2.31.1

